lib/ddtrace/contrib/cucumber/patcher.rb

```diff
@@ -16,6 +16,8 @@ def target_version
 
         def patch
           ::Cucumber::Runtime.send(:include, Instrumentation)
+          # force blocking flush before at_exit shutdown! hook
+          at_exit { Datadog.tracer.writer.worker.flush_data }
```

Comment:
It look like the hook in https://github.com/DataDog/dd-trace-rb/blob/10592cebfed7d4f9c509910e4d11faaf00cca668/lib/ddtrace.rb#L37 will always run before and it's too late to do this.

Unfortunately cucumber doesn't have a global exit hook like rspec. Any ideas?

Reply:
If we don't have a clear and fast way to fix this I will just remove the change from cucumber as our priority right now is fixing the rspec instrumentation

Reply:
Have you tried to hook into `TestRunFinished` https://github.com/cucumber/cucumber-ruby/blob/d962d8945a8249ef369238e5bc4ed20960e3c955/lib/cucumber/events/test_run_finished.rb#L7 ?